lib/thrift/Kconfig

```diff
@@ -14,12 +14,23 @@ config THRIFT
   help
     Enable this option to support Apache Thrift
 
+if THRIFT
+
+config THRIFT_SSL_SOCKET
+  bool "Enable TSSLSocket support for Thrift"
+  depends on MBEDTLS
+  depends on MBEDTLS_PEM_CERTIFICATE_FORMAT
+  depends on NET_SOCKETS_SOCKOPT_TLS
+  help
+    Enable this option to support TSSLSocket for Thrift
+    
 config THRIFT_ZLIB_TRANSPORT
   bool "Enable TZlibTransport support for Thrift"
-  depends on THRIFT
   help
     Enable this option to support TZlibTransport for Thrift
 
```

Comment:
I think there are 2 port variables that don't need to be static. Not a huge deal though. 

LGTM